modules/preview-react/segmented-control/lib/SegmentedControlList.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+
+import {
+  createSubcomponent,
+  ExtractProps,
+  createElemPropsHook,
+} from '@workday/canvas-kit-react/common';
+import {Stack} from '@workday/canvas-kit-react/layout';
+import {useListRenderItems} from '@workday/canvas-kit-react/collection';
+import {useSegmentedControlModel} from './hooks/useSegmentedControlModel';
+
+export interface SegmentedControlListProps<T = any>
+  extends Omit<Partial<ExtractProps<typeof Stack, never>>, 'children'> {
+  'aria-label': string;
+  children: ((item: T) => React.ReactNode) | React.ReactNode;
+}
+
+const useSegmentedControlList = createElemPropsHook(useSegmentedControlModel)(
+  ({state: {orientation, disabled}}) => {
+    return {
+      flexDirection: orientation === 'vertical' ? 'column' : 'row',
+      opacity: disabled ? 0.4 : 1,
```

Comment:
Preference here but do we need this hook? We can also accomplish this below in the style props too right?

```
opacity={model.state.disabled ? '0.4' : 1}
```

Honestly they're both valid ways to do thing! The power of hooks

Reply:
I would prefer to have hooks to apply all styles depending on model state values. Just to separate regular styles and state-specific.

Reply:
The `model` is available to `createSubcomponent` and `createContainer` if it is needed - mostly if the component MUST represent more than one element. It is rare, but valid. Ideally, we don't use `model` inside a component. That's what the `elemPropsHook` is for. Exporting the `elemPropsHook` makes it easier for developers to build their own component if necessary.